corehq/pillows/case_search.py

```diff
@@ -0,0 +1,151 @@
+    """
+    return not inspect.isdatadescriptor(getattr(CommCareCase, prop, None)) and re.search(r'^[a-zA-Z]', prop)
+
+
+class CaseSearchPillowProcessor(ElasticProcessor):
+    def process_change(self, pillow_instance, change, do_set_checkpoint):
+        domain = self._get_domain(change)
+        change_object = self._get_change_object(change)
+        if domain and case_search_enabled_for_domain(domain):
+            super(CaseSearchPillowProcessor, self).process_change(
+                pillow_instance, change_object, do_set_checkpoint
+            )
+
+    def _get_domain(self, change):
+        if isinstance(change, dict):
+            return change.get('key', [])[0]
+        elif isinstance(change, Change):
+            return change.metadata.domain
+
+    def _get_change_object(self, change):
+        if isinstance(change, dict):
+            return change_from_couch_row(change)
+        elif isinstance(change, Change):
+            return change
+
+
+def get_case_search_to_elasticsearch_pillow(pillow_id='CaseSearchToElasticsearchPillow'):
+    checkpoint = PillowCheckpoint(
+        'case-search-to-elasticsearch',
+    )
+    case_processor = CaseSearchPillowProcessor(
+        elasticseach=get_es_new(),
+        index_meta=ElasticsearchIndexMeta(index=CASE_SEARCH_INDEX, type=CASE_ES_TYPE),
+        doc_prep_fn=transform_case_for_elasticsearch
+    )
+    return ConstructedPillow(
+        name=pillow_id,
+        document_store=None,
+        checkpoint=checkpoint,
+        change_feed=KafkaChangeFeed(topics=[topics.CASE], group_id='cases-to-es'),
+        processor=case_processor,
+        change_processed_event_handler=PillowCheckpointEventHandler(
+            checkpoint=checkpoint, checkpoint_frequency=100,
+        ),
+    )
+
+
+def get_couch_case_search_reindexer(domain=None):
+    # TODO: Figure out how to not fetch every single case from the DB when running a full reindex
+
+    # TODO: Remove this
+    # It initializes the es index if it doesn't exist
+    CaseSearchPillow()
+
+    view_kwargs = {'include_docs': True}
+
+    if domain is not None:
+        if not case_search_enabled_for_domain(domain):
+            raise CaseSearchNotEnabledException("{} does not have case search enabled".format(domain))
+
+        startkey = [domain]
+        endkey = [domain, {}, {}]
+        view_kwargs.update({'startkey': startkey, 'endkey': endkey})
+
+    return PillowReindexer(get_case_search_to_elasticsearch_pillow(), CouchViewChangeProvider(
+        document_class=CommCareCase,
+        view_name='cases_by_owner/view',
+        view_kwargs=view_kwargs,
+    ))
+
+
+def delete_case_search_cases(domain):
+    if domain is None or isinstance(domain, dict):
+        raise TypeError("Domain attribute is required")
+
+    startkey = [domain]
+    endkey = [domain, {}, {}]
+    es = get_es_new()
+    db = CommCareCase.get_db()
+
+    for item in paginate_view(db, 'cases_by_owner/view', 100, startkey=startkey, endkey=endkey, reduce=False):
```

Comment:
I don't understand why this needs to access couch at all. can't you just run the domain-filtered query in the es index and delete all the results?